ocaml/xapi/helpers.ml

```diff
@@ -42,21 +42,15 @@ let log_exn_continue msg f x =
 type log_output = Always | Never | On_failure
 
 let filter_patterns =
-  [
-    ( Re.Str.regexp "^\\(.*proxy_\\(username\\|password\\)=\\)\\(.*\\)$"
-    , "\\1(filtered)"
-    )
-  ]
+  [(Re.Pcre.regexp "^(.*proxy_(username|password)=)(.*)$", "(filtered)")]
 
 let filter_args args =
   List.map
     (fun arg ->
       List.fold_left
         (fun acc (r, t) ->
-          if Re.Str.string_match r acc 0 then
-            Re.Str.replace_matched t acc
-          else
-            acc
+          try String.concat "" [(Re.Pcre.extract ~rex:r acc).(1); t]
```

Comment:
There should be an API that doesn't rely on exceptions, maybe in Re itself

Reply:
There is a `pmatch` function.

Reply:
Here I am able to do "does it match, and if so, return matching groups" in one go. Switching to `pmatch` means we would take two matches to do the same.

Reply:
ping @edwintorok